great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -108,7 +102,7 @@ def __init__(
         *,
         name: str,
         config_version: float,
-        rules: Dict[str, RuleConfig],
+        rules: Dict[str, CommentedMap],
```

Comment:
What should really go in here is a Dict of [str, Rule]; this might be a simple refactor; the logic to create the rule from config could be encapsulated in the rule or a builder method on rule config

Reply:
@donaldheppner We need to discuss the best way of accommodating this.  While I agree with this suggestion (and this goes back to the infamous effort a couple of years ago), our reliance on `instantiate_class_from_config()` is not conducive to this -- very proper in my opinion -- canonical approach.